src/subscription.rs

```diff
@@ -199,8 +203,22 @@ impl<T: Runtime> EventStorageSubscription<T> {
     /// Gets the next change_set from the subscription.
     pub async fn next(&mut self) -> Option<StorageChangeSet<T::Hash>> {
         match self {
-            Self::Imported(event_sub) => event_sub.next().await,
+            Self::Imported(event_sub) => read_subscription_response(event_sub).await,
             Self::Finalized(event_sub) => event_sub.next().await,
         }
     }
 }
+
+async fn read_subscription_response<T>(sub: &mut Subscription<T>) -> Option<T>
+where
+    T: DeserializeOwned,
+{
+    match sub.next().await {
```

Comment:
Note: this is a breaking change in `jsonrpsee` before all failures were ignored and this would only return on valid responses on the `subscription stream`.

We could change this API in `subxt` to send back the error here but I didn't want to make any breaking changes.

This error should only occur if the server doesn't send back a response on the stream that is the `type T` on the subscription.